File: solutions/ruby/hamming/1/hamming.rb
class Hamming
  def self.compute(strand1, strand2)
    # 1. Validación: las hebras deben tener la misma longitud
    if strand1.length != strand2.length
      raise ArgumentError, "Strands must be of equal length"
    end

    # 2. Convertimos el primer string a caracteres
    # 3. Usamos 'zip' para emparejarlos con el segundo string
    # 4. Contamos cuántas parejas (a, b) son distintas
    strand1.chars.zip(strand2.chars).count do |char1, char2|
      char1 != char2
    end
  end
end

File: solutions/r/hamming/1/hamming.R
hamming <- function(strand1, strand2) {
  # Check if the strands have the same length
  if (nchar(strand1) != nchar(strand2)) {
    stop("strands must be of equal length")
  }
  
  # Split the strings into individual characters
  chars1 <- strsplit(strand1, "")[[1]]
  chars2 <- strsplit(strand2, "")[[1]]
  
  # Count the number of differences
  sum(chars1 != chars2)
}